server/bundler/assets-writer.js

```diff
@@ -85,7 +85,9 @@ Object.assign( AssetsWriter.prototype, {
 				} )
 			);
 
-			self.outputStream.write( JSON.stringify( statsToOutput, null, '\t' ) );
+			self.outputStream.write( JSON.stringify( statsToOutput, null, '\t' ), () =>
```

Comment:
this can be an `.end( .. )` call since it's the last thing we write. That should catch any double writes.